### PR TITLE
eng_sql: add caption to kable html output

### PR DESCRIPTION
**Pull request by jjallaire**
Not sure if this is the right way to go about adding a caption to a kable table but here's my attempt.

### CODE REVIEW

**Comment by yihui** [comment]
We can just pass a caption to `kable(data, caption = ...)`. By default, use the caption you composed (n records), but users should be able to either suppress the caption (`NA`), or provide their own captions.

**Comment by jjallaire** [comment]
Okay, updated to pass `caption` directly to kable.

File: R/engine.R [hunk 1]
```diff
@@ -442,13 +442,28 @@ eng_sql = function(options) {
...
+        rows_formatted <- formatC(rows, format="d", big.mark=',')
+        caption <- if (max.print == -1 || rows < max.print)
+          paste(rows_formatted, "records")
+        else
+          paste("Displaying records 1 -", rows_formatted)
+      }
 
       # print using kable
-      print(kable(data))
+      print(kable(data, caption = caption, align = "l"))
```
Comment (yihui) [inline R/engine.R]: Is there a reason for left alignment?

**Comment by jjallaire** [comment]
What I really want to do is ensure that the first column is never right aligned (I don't care so much about the other columns). The issue with the first column is that for database queries it's very often the id field (numeric) which then ends up with right alignment by default. This in turn looks very strange in a table as the contents of the first column don't align with the left edge of the column.

So I guess really what I want is to use the default alignment behavior but then fixup the first column. I can easily do this via JS or CSS on the rmarkdown side but I wanted to do something that gave the same benefit to other renderings (e.g. LaTex, Word, etc.).

Try this demo with and without the align option to see what I am talking about:

    ```{r message=FALSE}
    library(DBI)
    db <- dbConnect(RSQLite::SQLite(), dbname = ":memory:")
    ```

    ```{sql, connection=db}
    CREATE TABLE packages (id INTEGER, name TEXT)
    ```

    ```{sql, connection=db}
    INSERT INTO packages VALUES (1, 'readr'), (2, 'readxl'), (3, 'haven'), (4, 'readr'), (5, 'readxl')
    ```

    ```{sql connection=db, max.print = 5}
    SELECT * FROM packages
    ```

**Comment by jjallaire** [comment]
I guess I could also just do an as.character on the first column before calling kable

**Comment by jjallaire** [comment]
Okay, I dropped the align="l" and instead am just converting the first column to character to force left alignment. Let me know if you think this is a reasonable approach.

**Comment by yihui** [comment]
It makes sense to left align the first column is it is the ID column, but I wonder if users doing data analysis will select the ID column from the database at all (I think the numeric ID column is rarely useful unless they want to join the data frame with other data frames later). The small 2-column example is kind of an extreme case: typically I don't think users only want to select 2 columns (and one of them is ID). I think the weird appearance of this table is mostly due to the 100% width showing too few columns.

**Comment by yihui** [comment]
It will also be strange if the first column is not incremental IDs but a numeric data column. I think a more robust approach is to actually check if it is an incremental column (e.g. `all(diff() == 1)`). That should significantly reduce the false positive rate.

**Comment by jjallaire** [comment]
Okay, I think you are correct on all counts (my example is a bit
contrived). I just updated the PR to only do as.character for all(diff() ==
1

**Pull request merged**